R/save_data.R

```diff
@@ -0,0 +1,58 @@
+#' Save extracted data
+#'
+#' Saves any extracted data as separate sheet of an xlsx file.
+#'
+#' @param import The import object to save the extracted data for.
+#' @param file Optional file to save the data to, defaults to creating file in
+#' current working directory called extracted_data.xlsx
+#'
+#' @export
+#'
+#' @examples
+#' import <- new_import(system.file("examples/person_information", package = "dettl"), "test")
+#' run_import(import, "extract")
+#' t <- tempfile()
+#' save_extracted_data(import, t)
+save_extracted_data <- function(import, file = "extracted_data.xlsx") {
+  save_data(import, file, "extracted")
+}
+
+#' Save transformed data
+#'
+#' Saves any transformed data as separate sheet of an xlsx file.
+#'
+#' @param import The import object to save the transformed data for.
+#' @param file Optional file to save the data to, defaults to creating file in
+#' current working directory called transformed_data.xlsx
+#'
+#' @export
+#'
+#' @examples
+#' import <- new_import(system.file("examples/person_information", package = "dettl"), "test")
+#' run_import(import, c("extract", "transform))
+#' t <- tempfile()
+#' save_transformed_data(import, t)
+save_transformed_data <- function(import, file = "transformed_data.xlsx") {
+  save_data(import, file, "transformed")
+}
+
+save_data <- function(import, file, stage) {
+  if (!inherits(import, "dataImport")) {
+    stop(sprintf("Can't save %s data for non-DataImport object.", stage))
+  }
+  if (!file.exists(file)) {
```

Comment:
what is the role of the file create here?  the writexl package doesn't need this surely?  You might do (unconditionally) `dir.create(dirname(file), FALSE, TRUE)` if it's just the directory tree you want to ensure exists

Reply:
Yep I just want to make sure the directory exists.